### PR TITLE
[task #8213]Port tests in select.rs to sqllogictest

### LINKED ISSUE #8213: Port tests in `select.rs` to sqllogictest
### Is your feature request related to a problem or challenge?

Part of https://github.com/apache/arrow-datafusion/issues/6195. We are trying to
1. keep the DataFusion code base clean and easy to contribute to and modify
2. Understand what features are covered and what are not

Part of doing so is having a single location for most test coverage so new tests can be added easily and people can follow the existing patterns easily

### Describe the solution you'd like

Port the tests in https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/tests/sql/select.rs to
`select.slt` in https://github.com/apache/arrow-datafusion/tree/main/datafusion/sqllogictest/test_files

Note you can create catalogs and schema via SQL now

Catalogs: https://arrow.apache.org/datafusion/user-guide/sql/ddl.html#create-database
Schema: https://arrow.apache.org/datafusion/user-guide/sql/ddl.html#create-schema

Notes:
1. Instructions for running sqllogitest are here: https://github.com/apache/arrow-datafusion/tree/main/datafusion/sqllogictest

### Describe alternatives you've considered

_No response_

### Additional context

I think these are good first issues as they teach the contributor about the DataFusion codebase and tests, as well as see the end user apis in practice.

### PR DESCRIPTION
**Pull request by Tangruilin**
## Which issue does this PR close?

Closes #8213
Closes https://github.com/apache/arrow-datafusion/pull/8939
Closes https://github.com/apache/arrow-datafusion/pull/8953
.

### CODE REVIEW

**Review by comphead — APPROVED** [review]
lgtm thanks @Tangruilin

**Comment by Tangruilin** [comment]
image
for the sql `PREPARE my_plan(INT, DOUBLE) AS SELECT c1, c2 FROM test WHERE c1 > $2 AND c1 < $1`

**Comment by Tangruilin** [comment]
> image for the sql `PREPARE my_plan(INT, DOUBLE) AS SELECT c1, c2 FROM test WHERE c1 > $2 AND c1 < $1`

There is a error for the sql, and others I have all solved.
You can review it.
@alamb

**Comment by comphead** [comment]
Tests are failing, and btw there is also another PR https://github.com/apache/arrow-datafusion/pull/8953

**Comment by Tangruilin** [comment]
> Tests are failing, and btw there is also another PR #8953

I have solved the failed, you can review it again

**Comment by Tangruilin** [comment]
@alamb This CR is ready to be review

File: datafusion/sqllogictest/test_files/select.slt [hunk 1]
```diff
@@ -46,11 +46,300 @@ STORED AS CSV
...
+(3, 4, true),
+(3, 5, false),
+(3, 6, true),
+(3, 7, false),
+(3, 8, true),
+(3, 9, false);
+
+# test_list_query_parameters
+query IIB
+SELECT * FROM query_parameters_table WHERE c1 = 3;
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/select.slt]: I think this is different than the original test where the parameter is filled in via `with_param_value`. I will restore it back to the .rs file

```
        .sql("SELECT * FROM test WHERE c1 = $1")
```

**Review by alamb — APPROVED** [review]
Thank you @Tangruilin and @simicd  -- I went through this carefully and the only issue I found was that the query parameter test I think can't be written in SQL so needs to part of `.rs`

Since this PR has dragged out a bunch already, I made the adjustment myself and I plan to merge this PR once the CI passes. 

I am really sorry @simicd that we didn't see / prevent this from being implemented in parallel to https://github.com/apache/arrow-datafusion/pull/8953 and  https://github.com/apache/arrow-datafusion/pull/8939 

I will make a PR to the contributor guide to try and make the expectations clearer and hopefully avoid this kind of repetition in the future

**Comment by alamb** [comment]
Thanks again everyone

**Pull request merged**